--- linux-3.0.8/drivers/mtd/devices/hisfc350/hisfc350.c 2019-07-22 14:48:08.462002879 +0200
+++ linux-3.0.8/drivers/mtd/devices/hisfc350/hisfc350.c 2019-07-22 14:49:04.374696679 +0200
@@ -1173,7 +1173,7 @@

 static struct platform_driver hisfc350_driver_pltdrv = {
        .probe          = hisfc350_driver_probe,
-       .remove         = hisfc350_driver_remove,
+       .remove         = __devexit_p(hisfc350_driver_remove),
        .shutdown       = hisfc350_driver_shutdown,
 #ifdef CONFIG_PM
        .suspend        = hisfc350_driver_suspend,
