From: Claus Marxmeier <alladin@routeme.de>
Date: Sa, 21 Jul 2019 02:24:19 +0200
Subject: [PATCH] cfg80211: reuse existing page fragments in A-MSDU rx - Kernerl Version check

added Kernel-Version check to get build with Kernel 3.0.8

Signed-off-by: Claus Marxmeier <alladin@routeme.de>
---


--- a/net/wireless/util.c  2019-07-21 02:18:00.303302041 +0200
+++ b/inet/wireless/util.c  2019-07-21 02:19:39.748557471 +0200
@@ -746,7 +746,11 @@
        u8 *payload;
        int offset = 0, remaining, err;
        struct ethhdr eth;
+#if LINUX_VERSION_IS_LESS(3,5,0)
+       bool reuse_frag = 0;
+#else
        bool reuse_frag = skb->head_frag && !skb_has_frag_list(skb);
+#endif
        bool reuse_skb = false;
        bool last = false;

